app/src/main/java/com/google/droidjump/GameView.java

```diff
@@ -16,16 +16,156 @@
 
 package com.google.droidjump;
 
+import static androidx.navigation.Navigation.findNavController;
+import android.annotation.SuppressLint;
 import android.content.Context;
+import android.graphics.BitmapFactory;
+import android.graphics.Canvas;
+import android.graphics.Color;
+import android.view.MotionEvent;
+import android.view.SurfaceHolder;
 import android.view.SurfaceView;
 
 public class GameView extends SurfaceView implements Runnable {
 
+    private boolean isPlaying;
+    private int screenX;
+    private int screenY;
+    private int screenMargin;
+    private SurfaceHolder surfaceHolder;
+    private MainActivity activity;
+    private Droid droid;
+    private int timePoint;
+    Thread thread;
+    private int levelTimePoints;
+    private int levelSpeed;
+
     public GameView(Context context) {
         super(context);
+        isPlaying = false;
+    }
+
+    public GameView(Context context, int screenX, int screenY, boolean isPlaying) {
+        super(context);
+        timePoint = 0;
+        surfaceHolder = getHolder();
+        activity = (MainActivity) context;
+        this.screenX = screenX;
+        this.screenY = screenY;
+        this.isPlaying = isPlaying;
+
+        // Margin in px
+        screenMargin = (int) getResources().getDimension(R.dimen.fab_margin);
+        // Measuring droid jump height
+        DrawableElement palm = new DrawableElement(/* x= */0, /* y= */0,
+                /* bitmap= */BitmapFactory.decodeResource(getResources(), R.mipmap.palm));
+        int jumpHeight = palm.getHeight() + 100;
+        // Create droid
+        droid = new Droid(screenMargin, screenY - screenMargin, jumpHeight, getResources());
     }
 
     @Override
     public void run() {
+        receiveLevelDetails();
+        draw();
+        while (isPlaying) {
+            update();
+            draw();
+            sleep();
+            timePoint++;
+        }
+    }
+
+    private void receiveLevelDetails() {
+        // TODO: Serialize current level data and put it in some container
+        levelTimePoints = 200;
+        levelSpeed = 3;
```

Comment:
What is the "levelSpeed" for? You don't use it anywhere.

Reply:
This will be used later after receiving level data(Speed of moving ground, obstacles from right to the left side of the screen). Should I remove it for now?